src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
+        int newLength = (int) (size * 1.5);
+
+        if (size == array.length) {
+            array = resize(newLength);
+        }
 
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        T[] newArray = (T[]) new Object[array.length];
 
+        if (index < 0) {
+            throw new ArrayIndexOutOfBoundsException(index);
+        }
```

Comment:
what if index > size?

Reply:
I'll resize array and put new value to the requested index